util/helm/helm.go

```diff
@@ -28,30 +31,89 @@ type helm struct {
 	path string
 }
 
-func (h *helm) Template(name string, valuesFiles []string, params []*argoappv1.ComponentParameter) ([]*unstructured.Unstructured, error) {
+func (h *helm) Template(name string, valuesFiles []string, overrides []*argoappv1.ComponentParameter) ([]*unstructured.Unstructured, []*argoappv1.ComponentParameter, error) {
```

Comment:
Can we split Helm object to have two functions which serve different purposes?

Template() renders the chart, and mirrors the functionality of `helm template`
GetParameters() returns the available parameters 

This will make the library very similar to our Ksonnet library.

Reply:
Good suggestion @jessesuen . Done